src/main/java/nearsoft/academy/bigdata/recommendation/MovieRecommender.java

```diff
@@ -0,0 +1,169 @@
+    model = new FileDataModel(new File("data.csv"));
+    //Threshold Similarity:
+    similarity = new PearsonCorrelationSimilarity(model);
+    neighborhood = new ThresholdUserNeighborhood(0.1, similarity, model);
+    // Create Recommender
+    this.recommender = new GenericUserBasedRecommender(model, neighborhood, similarity);
+    
+  }
+  
+  /**
+   * <p> Generates .csv file in root folder from a .txt file.</p>
+   * @param path path of the .txt file with amazon reviews.
+   * @throws IOException recommender handles IOExceptions.
+   * @since 1.0
+   * */
+  
+  private void csvBuilder(String path) throws IOException {
+    File file = new File(path);
+    BufferedReader br = new BufferedReader(new FileReader(file));
+    FileWriter writer = new FileWriter("data.csv");
+    BufferedWriter bw = new BufferedWriter(writer);
+    // Iterate through .txt to create each line of the new .csv file
+    for (String line = br.readLine(); line != null; line = br.readLine()) {
+      
+      if (line.startsWith("review/score")) {
+        String reviewScore = line.split(" ")[1];
+        bw.write(this.userPortion + this.productPortion + reviewScore + "\n");
+      } else {
+        // Call method to check for the remaining components of .csv line
+        buildCsvLine(line);
+      }
+      
+    }
+    
+    bw.close();
+  }
+  
+  /**
+   * <p>Inserts values in "productPortion" and "userPortion" for new .csv line.</p>
+   * @param line current line in .txt file.
+   * @since 1.0
+   * */
+  
+  private void buildCsvLine(String line) {
+    if (line.startsWith("product/productId")) {
+      // Read new product. Create new csv line
+      String productId = line.split(" ")[1];
+      this.totalReviews++;
+    
+      if (this.productsMap.containsKey(productId)) {
+        // Use the key in hash map
+        this.productPortion = this.productsMap.get(productId) + ",";
+      } else {
+        // Add new key to hash maps
+        this.productsMap.put(productId, this.totalProducts);
+        this.invertedProductsMap.put(this.totalProducts, productId);
+        this.productPortion = this.totalProducts + ",";
+        this.totalProducts++;
+      }
+    } else if (line.startsWith("review/userId")) {
+      String userId = line.split(" ")[1];
+    
+      if (this.usersMap.containsKey(userId)) {
+        // Use the key in hash map
+        this.userPortion = this.usersMap.get(userId) + ",";
+      } else {
+        // Add new key to hash map
+        this.usersMap.put(userId, this.totalUsers);
+        userPortion = this.totalUsers + ",";
+        this.totalUsers++;
+      }
+    }
+  }
+  
+  
+  /**
+   * <p>Returns 3 product recommendations for an specific user.</p>
+   * @param userId user that we are recommending 3 products.
+   * @return List containing Strings of product ids.
+   * @throws TasteException recommender handles TasteExceptions
+   * @since 1.0
+   * */
+  
+  List<String> getRecommendationsForUser(String userId) throws TasteException {
```

Comment:
you feel ok with the throwing of the exception? , instead of having a try-catch and always return a list of results?

Reply:
Yes, since it's a very small portion of code that does only one task specifically I would rather have the caller handle the exception instead of my method getRecommendationsForUser.
But since right now there is no method that calls this function (and also can be accessed by the class and package) I will change it for a try-catch so that the method handles the exception. Thanks for pointing it out Julio!